rust/segment/src/local_hnsw.rs

```diff
@@ -624,6 +601,38 @@ impl LocalHnswSegmentWriter {
                 }
             }
         }
+
+        // Add to hnsw index in parallel using rayon.
+        // Resize the index if needed
+        let index_len = guard.index.len_with_deleted();
+        let index_capacity = guard.index.capacity();
+        if index_len + hsnw_batch.len() >= index_capacity {
+            guard
+                .index
+                .resize(index_capacity * 2)
```

Comment:
can this not be enough? For e.g. if hnsw_batch.len() > 2 * index_capacity

Reply:
woops thanks